### DISCUSSION TITLE
Better Lua replacements

### DISCUSSION THREAD

**Discussion by data-man** [dail8859/NotepadNext#396] [Ideas]
[Luau](https://luau-lang.org) ([sources](https://github.com/Roblox/luau)) is the fast Lua alternative, written on C++17 with many useful additions.
BTW [LuaBridge3](https://github.com/kunitoki/LuaBridge3) supports Luau out-of-the-box.

**Comment by dail8859** [comment]
I had originally dropped in Lua from one of my Notepad++ plugins mainly for some debugging and flexibility during development. I've started relying on Lua a bit more than I had originally planned but not so much it couldn't be replaced/removed.

I wouldn't be opposed to something like Luau/LuaBridge3 especially since it seems to be backwards compatible (or at least easily enough). I don't foresee that there would be any noticeable performance gains since Lua isn't _currently_ being used in any type of critical code path. However, using something that is improved/supported/modern is never a bad thing.

**Reply by data-man** [reply]
> I wouldn't be opposed to something like Luau/LuaBridge3

So I can start working on that, if you don't mind. But I think we should switch to CMake first. :)

**Reply by dail8859** [reply]
CMake is also desirable. Work was started by another user in #100 but not finished. I am not very familiar with CMake and currently don't have time to try to learn it from scratch.

**Comment by data-man** [comment]
https://pluto-lang.org looks even better. :)
